### DISCUSSION TITLE
Meta: Creating A Mobile App

### DISCUSSION THREAD

**Discussion by detherminal** [cockpit-project/cockpit#18560] [Ideas]
Hello, I am an android developer who codes things for learning and fun. I would like to make a mobile app for Cockpit under the name "Cockpit Mobile" and I would like to publish the app on Google Play Store on my personal account. I want permission from the Cockpit team so I can work on it.

What is your ideas and can you give me permission?

**Comment by martinpitt** [comment]
Can you please be more specific what you mean with "mobile app"? Cockpit already supports mobile browsers rather well, we put quite a lot of effort into it. I.e. you can point the browser on your phone to https://some.server:9090 and you should get something useful. But I figure that's not what you mean.

An interesting project might be to have an Android app counterpart of our [Cockpit Client flatpak](https://flathub.org/apps/details/org.cockpit_project.CockpitClient)  -- i.e. bundling the web server, login page, and cockpit-ssh so that you can point it to any machine and just have SSH installed there, instead of cockpit-ws. If you have that in mind, then by all means go ahead!

FWIW, you don't need our "permission"..  If you bundle Cockpit code in your app, it must be compliant with the LGPL v2, but other than that, this is free software :grin:

**Reply by Victorouma57** [reply]
hello  sir , am victor ouma omwanda a kenyan who has an interest in mobile  app  please i want a person who is an expert in doing that so  that i be guided i will be willing to pay for the token of training. my email is victorouma57@gmail.com

**Comment by detherminal** [comment]
I am thinking of a native application for Android that has good ui and ux. Browser apps are good but not good enough to be used daily on phone. So can I make it under the name "Cockpit Mobile"? If I can I am gonna start writing a mobile app client for Cockpit. I hope I dont give up easily. 😄

**Comment by ragavpr** [comment]
I can think about some important stuff that an app can do, but the browser can't
- Persistent notifications, that can show some status which we customized to show
- Alarm notifications, say in case of any alert (high cpu / mem, high temp / etc.)

**Comment by User8395** [comment]
I'm working on one right now. It uses a WebView to embed the Cockpit UI.

**Reply by SpvceR3ii** [reply]
Is this still ongoing? If so, can I be of any help?

I'm an AI developer, but most importantly, I know a lot of coding languages. 

I used to code with React Native, but I still have a sharp mind if you need help.

**Comment by Jaixxer** [comment]
Guys, If this is still going on, I would love to be a part of this. Can contribute a ton. If it's not, do tell, so I may start from scratch.

**Reply by Venefilyn** [reply]
None of the replies have reported any progress so I'd assume no one has created anything

**Reply by Jaixxer** [reply]
Oh okay
